components/Buttons/src/MDCButton.m

```diff
@@ -890,7 +891,13 @@ - (void)updateTitleFont {
   if (_mdc_adjustsFontForContentSizeCategory) {
     // Dynamic type is enabled so apply scaling
     if (font.mdc_scalingCurve && !_mdc_legacyFontScaling) {
-      font = [font mdc_scaledFontForCurrentSizeCategory];
+      UIContentSizeCategory sizeCategory = UIContentSizeCategoryLarge;
+      if (@available(iOS 10.0, *)) {
+        sizeCategory = self.traitCollection.preferredContentSizeCategory;
+      } else if ([UIApplication mdc_safeSharedApplication]) {
+        sizeCategory = [UIApplication mdc_safeSharedApplication].preferredContentSizeCategory;
+      }
+      font = [font mdc_scaledFontForSizeCategory:sizeCategory];
     } else {
       font = [font mdc_fontSizedForMaterialTextStyle:MDCFontTextStyleButton
                                 scaledForDynamicType:YES];
```

Comment:
Does the "double-set" hack need to be applied on line 907?

Reply:
I don't believe so.